arangod/Aql/QueryContext.cpp

```diff
@@ -56,15 +57,18 @@ QueryContext::QueryContext(TRI_vocbase_t& vocbase)
       _vocbase(vocbase),
       _execState(QueryExecutionState::ValueType::INVALID_STATE),
       _numRequests(0) {
-  if (!AqlFeature::lease()) {
+  if (&_vocbase != &_vocbase.server().getFeature<DatabaseFeature>().getCalculationVocbase() && 
```

Comment:
Is this a performance optimization only, or are there any other reasons to allow calculations to be executed during shutdown? This is unclear from the code alone, so we should add a code comment here.

Reply:
added comments to make intentions clear